compiler/rustc_mir_transform/src/pass_manager.rs

```diff
@@ -121,7 +121,7 @@ fn run_passes_inner<'tcx>(
                 validate_body(tcx, body, format!("before pass {}", name));
             }
 
-            pass.run_pass(tcx, body);
+            tcx.sess.time(name, || pass.run_pass(tcx, body));
```

Comment:
This should be using `generic_activity` or `generic_activity_with_arg` to avoid printing lots of MIR passes on stderr.